compiler/rustc_lint/messages.ftl

```diff
@@ -362,6 +362,10 @@ lint_impl_trait_redundant_captures = all possible in-scope parameters are alread
 
 lint_implicit_unsafe_autorefs = implicit autoref creates a reference to the dereference of a raw pointer
     .note = creating a reference requires the pointer target to be valid and imposes aliasing requirements
+    .raw_ptr = this raw pointer has type `{$raw_ptr_ty}`
+    .autoref = autoref is being applied to this expression, resulting in: `{$autoref_ty}`
+    .through_overloaded_deref = reference(s) created through call(s) to overloaded `Deref(Mut)::deref(_mut)` implementation
```

Comment:
```suggestion
    .through_overloaded_deref = references are created through calls to explicit `Deref(Mut)::deref(_mut)` implementations
```

1. Don't use term "overloaded" in user-facing diagnostics. I've used "explicit" but choose whatever else sounds more natural
2. By making it a general statement you can get rid of these slightly hard to read `(s)`s

Reply:
Yeah, "overloaded deref" is more of a technical term, let's be more generic. :+1: